src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -345,6 +352,13 @@ export class SavedObjectsService
         }
         this.typeRegistry.registerType(type);
       },
+      registerExportHook: (type, hook) => {
+        if (this.started) {
+          throw new Error('cannot call `registerExportHook` after service startup.');
+        }
+        // TODO
+        this.exportHooks.set(type, hook);
+      },
```

Comment:
As `registerExportHook` is dissociated from `registerType`, it is theoretically possible to register multiple hooks for the same type. However this would create a lot of complexity (order of execution, recursivity...), so I think we should just forbid that and throw in that case?

Reply:
We should probably validate that the type has `management.importableAndExportable: true` if it specifies an export hook.